criu/files.c

```diff
@@ -548,7 +548,8 @@ static int dump_one_file(struct pid *pid, int fd, int lfd, struct fd_opts *opts,
 
 		p.link = &link;
 
-		if (is_memfd(p.stat.st_dev))
+		/* TODO: Dump for hugetlb fd when memfd hugetlb is not supported */
+		if (is_memfd(p.stat.st_dev) || (is_hugetlb_dev(p.stat.st_dev, NULL) && kdat.has_memfd_hugetlb))
```

Comment:
@minhbq-99 the order here is important. `is_hugetlb_dev()` is called when `kdat.has_memfd_hugetlb` is `0`.

Reply:
Same as the below comment